### PR TITLE
closeEvent & Restore Function

### PR DESCRIPTION
**Pull request by the0cp**
1. Update the closeEvent:
- Display a dialog with "Exit" and "Minimize" buttons upon clicking the "X", with "Exit" triggering the `onExit()` function and "Minimize" hiding the window to the system tray.
- Add a checkbox "don't ask me again".
2. Delete(comment) `changeEvent()`.
3. Add function to restore settings.

Here's the demo(mp4): 

https://github.com/AlexIII/tcc-g15/assets/63190431/686bd57e-3d80-4d8a-b6ee-f81406ae4f6b

### CODE REVIEW

File: src/Backend/__pycache__/AWCCThermal.cpython-312.pyc
Comment (AlexIII) [inline src/Backend/__pycache__/AWCCThermal.cpython-312.pyc:1]: Remove all the `__pycache__` files from the commit. Leave only *.py source files.

File: src/GUI/AppGUI.py [hunk 1]
```diff
@@ -68,6 +68,7 @@ class SettingsKey(Enum):
     CPUThresholdTemp = "app/fan/cpu/threshold_temp"
     GPUFanSpeed = "app/fan/gpu/speed"
     GPUThresholdTemp = "app/fan/gpu/threshold_temp"
+    exitValue = "app/exit"
```
Comment (AlexIII) [inline src/GUI/AppGUI.py:71]: Capitalize and rename to `MinimizeOnCloseFlag` (or choose other meaningful name)

File: src/GUI/AppGUI.py [hunk 2]
```diff
@@ -296,6 +299,7 @@ def updateOutput():
         self._loadAppSettings()
 
     def closeEvent(self, event):
+        '''
```
Comment (AlexIII) [inline src/GUI/AppGUI.py:302]: Remove this code if it's not needed.

File: src/GUI/AppGUI.py [hunk 3]
```diff
@@ -324,6 +373,15 @@ def _loadAppSettings(self):
...
 
+    def clearAppSettings(self):
+        msg_box = QtWidgets.QMessageBox(QtWidgets.QMessageBox.Warning, 'Restore Default',
+                                  "Do you want to restore all settings to default?", QtWidgets.QMessageBox.Yes |
+                                  QtWidgets.QMessageBox.No, self)
+        reply = msg_box.exec_()
+        if reply == QtWidgets.QMessageBox.Yes:
+            self.settings.clear()
+
+    '''
```
Comment (AlexIII) [inline src/GUI/AppGUI.py:384]: Remove this code if it's not needed

**Review by AlexIII — CHANGES_REQUESTED** [review]
Hi! Thank you for the contribution!

Please see the review.

**Comment by the0cp** [comment]
Thank you for your code review. I am new to PyQt. Sorry some details were not handled well. 
As a Dell G series user I'm glad you developed this software as an alternative to the **awful** AWCC

**Comment by AlexIII** [comment]
Accepted. The changes will be integrated to the master branch upon the next release.
Again, thank you for your contribution to the project!

**Pull request merged**